### PR TITLE
Feature/ab#23915 cas api

### PR DESCRIPTION
**Pull request by jimmyPasta**
Not tied into the  Approvals yet until we get the logic for L1/L2 Approvals completed

### CODE REVIEW

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/Invoice.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+using System.ComponentModel.DataAnnotations;
+
+
+namespace Unity.Payments.Integrations.Cas
+{
+    public class Invoice
+    {
+        [Required]
+        [MaxLength(25)]
+        public string? invoiceType { get; set; } = "Standard";
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/Invoice.cs:11]: lets update the Casing of these properties and use the json property name to serialize as needed pls

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/InvoiceLineDetail.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+namespace Unity.Payments.Integrations.Cas
+{
+    public class InvoiceLineDetail : IValidatableObject
+    {
+        [Required]
+        public int invoiceLineNumber { get; set; }
+
+        [Required]
+        [MaxLength(4, ErrorMessage = "invoiceLineType must be 'Item'")]
+        public string invoiceLineType { get; set; } = "Item";
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/InvoiceLineDetail.cs:13]: Pascal case and json property name for serialization name, as as above pls

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application/Domain/PaymentRequests/IPaymentRequestRepository.cs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 namespace Unity.Payments.Domain.PaymentRequests
 {
-    public interface IPaymentRequestRepository : IBasicRepository<PaymentRequest, Guid>
-    {
+    public interface IPaymentRequestRepository : IRepository<PaymentRequest, Guid>
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application/Domain/PaymentRequests/IPaymentRequestRepository.cs:6]: Please don't use the IRepository, we used it in the main app but we shouldn't trying to implement the modules as per best practices.
https://docs.abp.io/en/abp/latest/Best-Practices/Repositories

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application/Integration/Cas/InvoiceService.cs [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+            _iPaymentRequestRepository = paymentRequestRepository;
+            _paymentConfigurationAppService = paymentConfigurationAppService;
+            _resilientRestClient = resilientHttpRequest;
+            _casClientOptions = casClientOptions;
+            _iSupplierRepository = iSupplierRepository;
+            _iSiteRepository = iSiteRepository;
+            _restClient = restClient;
+        }
+
+        private async Task<Invoice?> InitializeCASInvoice(PaymentRequest paymentRequest,
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application/Integration/Cas/InvoiceService.cs:65]: Look at the best practices for application services - as above, trying to be good with implementation / best practices with all the new modules

https://docs.abp.io/en/abp/latest/Best-Practices/Application-Services

In particular:
Application Service Implementation
Do develop the application layer completely independent from the web layer.
Do implement application service interfaces in the application layer.
Do use the naming convention. Ex: Create ProductAppService class for the IProductAppService interface.
Do inherit from the ApplicationService base class.
Do make all public methods virtual, so developers may inherit and override them.
Do not make private methods. Instead make them protected virtual, so developers may inherit and override them.

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/Invoice.cs (hunk 1 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/Invoice.cs]: Done, CAS expects these as lower case and throws an exception for the cap case. Had to add decorator property to resolve this and move away from NewtonSoft JSON serialization. The code base I coppied this from must have had the same issue and made them lower case for conveniance of conversion.

File: applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/InvoiceLineDetail.cs (hunk 2 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application.Contracts/Integrations/Cas/InvoiceLineDetail.cs]: Done, CAS expects these as lower case and throws an exception for the cap case. Had to add decorator property to resolve this and move away from NewtonSoft JSON serialization. The code base I coppied this from must have had the same issue and made them lower case for conveniance of conversion.

**Pull request merged**